src/compiler/optimizer.js

```diff
@@ -17,13 +18,27 @@ let isPlatformReservedTag
  */
 export function optimize (root: ?ASTElement, options: CompilerOptions) {
   if (!root) return
+  isStaticKey = genStaticKeys(options.modules)
```

Comment:
How about generate a string option `staticKeys` directly in `CompilerOptions` in `web-compiler.js`?